### PR TITLE
Adds support for mapzen terrain

### PR DESCRIPTION
**Pull request by mollymerp**
continuing from @ibesora's work at  #6024 

@ibesora I rebased your PR and fixed the merge conflicts, and then I added a couple comments and tests 😊 . I don't have push access to your branch, so I made this new one (as I did with the maxzoom PR)

### CODE REVIEW

**Comment by ibesora** [comment]
Looks great! You guys have such a nice project structure. Great work!

**Comment by mollymerp** [comment]
@ibesora here is the gl-native equivalent https://github.com/mapbox/mapbox-gl-native/pull/11154e

File: src/data/dem_data.js [hunk 1]
```diff
@@ -57,22 +57,14 @@ class DEMData {
         this.loaded = !!data;
     }
 
-    loadFromImage(data: RGBAImage) {
+    loadFromImage(data: RGBAImage, encoding?: "mapbox" | "terrarium") {
```
Comment (anandthakker) [inline src/data/dem_data.js]: `encoding` can be non-optional here

File: src/source/raster_dem_tile_worker_source.js [hunk 2]
```diff
@@ -21,11 +21,12 @@ class RasterDEMTileWorkerSource {
     }
 
     loadTile(params: WorkerDEMTileParameters, callback: WorkerDEMTileCallback) {
-        const uid = params.uid;
+        const uid = params.uid,
+            encoding = params.encoding || "mapbox";
```
Comment (anandthakker) [inline src/source/raster_dem_tile_worker_source.js]: the `|| "mapbox"` shouldn't be needed since `WorkerDemTileParameters.encoding` is required.

**Pull request merged**